### PR TITLE
add NX_ARBITRARY_UNITS

### LINKED ISSUE #921: How to specify a field will have arbitrary units?
Propose adding `NX_ARBITRARY_UNITS` to the list of possible unit types.

### PR DESCRIPTION
**Pull request by prjemian**
- FIX #921
- REF #943

### CODE REVIEW

File: nxdlTypes.xsd [hunk 1]
```diff
@@ -102,6 +103,34 @@
...
+				units to some engineering units.  This should be considered as a
+				possibility in the analysis process.
+
+				**NOTE**: not the same as NX_UNITLESS or NX_DIMENSIONLESS
+
+				**NOTE**: It is common (such as in :ref:`NXcanSAS`) for intensity to
+				be reported in arbitrary units.
+
+				<xs:element name="example">arbitrary</xs:element>
+				<xs:element name="example">a.u.</xs:element>
```
Comment (PeterC-DLS) [inline nxdlTypes.xsd:128]: Not to be confused with astronomical units(!)

**Comment by phyy-nx** [comment]
Are these the same as ADU (analog digital units)? I always understood ADU as being an arbitrary unit.  If so, we could mark our detector data as in arbitrary units for systems where that's applicable.

**Comment by prjemian** [comment]
ADU usually has a scaling factor to engineering units

File: nxdlTypes.xsd (hunk 1 above)
Comment (prjemian) [reply nxdlTypes.xsd:128]: noted

**Comment by prjemian** [comment]
@phyy-nx The example of arbitrary units is the relative or normalized scattered intensity.  Often intensity is measured in counts.  Normalizing by a constant source parameter (such as incident beam monitor) can involving mixing of units (counts divided by photocurrent if the monitor is an ioc chamber, for example) where the conversion factor is too cumbersome to represent or no use of such a factor is made in data processing.  Hence: _arbitrary units_ have become customary.

**Comment by phyy-nx** [comment]
Great, thanks for the clarification

**Comment by prjemian** [comment]
Adding this new unit type is seen as unnecessary.  NeXus technical committee discussed possibilities using [`NX_UNITLESS`](https://github.com/nexusformat/definitions/blob/4e48fcf6a6d8e01bf995bd0ced4b4b416ddb6e80/nxdlTypes.xsd#L78), [`NX_DIMENSIONLESS`](https://github.com/nexusformat/definitions/blob/4e48fcf6a6d8e01bf995bd0ced4b4b416ddb6e80/nxdlTypes.xsd#L57), if the units truly cancel, or omitting units entirely.

In the end, the tech committee recommended using `units="unknown"` (which is recognized by UDUNITS) in such cases, rather than creating a new type of units.  Application software will recognize this value when considering the units of a field.

**Pull request closed without merging**